pyccel/ast/operators.py

```diff
@@ -106,7 +106,7 @@ class PyccelOperator(PyccelAstNode):
     args: tuple
         The arguments passed to the operator
     """
-    __slots__ = ('_args','_dtype','_precision','_shape','_rank','_order')
+    __slots__ = ('_args','_order', )
```

Comment:
`_order` can probably be treated the same way as `_dtype`, `_precision`, `_shape` and `_rank`. It tells us if an array is saved in Fortran ordering (column-major) or C ordering (row-major) (see [here](https://en.wikipedia.org/wiki/Row-_and_column-major_order) for more details). It should be set to `None` if `_rank`<2 so anywhere the `_rank` is a class property probably `_order` can be too

Reply:
I am happy to leave this improvement for another PR though as it may require significantly more code?